cmd/server.go

```diff
@@ -0,0 +1,227 @@
+The server arg is optional, if not passed in a list of available servers will
+be displayed and one can be selected.
+`,
+				Action: serverDelete,
+			},
+			{
+				Name:   "ls",
+				Usage:  "List all servers",
+				Action: serverLs,
+			},
+			{
+				Name:      "switch",
+				Usage:     "Switch to a new server",
+				ArgsUsage: "[SERVER_NAME]",
+				Description: `
+The server arg is optional, if not passed in a list of available servers will
+be displayed and one can be selected.
+`,
+				Action: serverSwitch,
+			},
+		},
+	}
+}
+
+// serverCurrent command to display the name of the current server in the local config
+func serverCurrent(ctx *cli.Context) error {
+	cf, err := loadConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	serverName := cf.CurrentServer
+	URL := cf.Servers[serverName].URL
+	fmt.Printf("Name: %s URL: %s\n", serverName, URL)
+	return nil
+}
+
+// serverDelete command to delete a server from the local config
+func serverDelete(ctx *cli.Context) error {
+	cf, err := loadConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	var serverName string
+	if ctx.NArg() == 1 {
+		serverName = ctx.Args().First()
+	} else {
+		serverName, err = serverFromInput(ctx, cf)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, ok := cf.Servers[serverName]
+	if !ok {
+		return errors.New("Server not found")
+	}
+
+	delete(cf.Servers, serverName)
+	cf.Write()
+	logrus.Infof("Server %s deleted", serverName)
+	return nil
+}
+
+// serverLs command to list rancher servers from the local config
+func serverLs(ctx *cli.Context) error {
+	cf, err := loadConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	writer := NewTableWriter([][]string{
+		{"CURRENT", "Current"},
+		{"NAME", "Name"},
+		{"URL", "URL"},
+	}, ctx)
+
+	defer writer.Close()
+
+	for name, server := range cf.Servers {
+		var current string
+		if name == cf.CurrentServer {
+			current = "*"
+		}
+		writer.Write(&serverData{
+			Current: current,
+			Name:    name,
+			URL:     server.URL,
+		})
+	}
+
+	return writer.Err()
+}
+
+// serverSwitch command to switch rancher server.
+func serverSwitch(ctx *cli.Context) error {
+	cf, err := loadConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	var serverName string
+	if ctx.NArg() == 1 {
+		serverName = ctx.Args().First()
+	} else {
+		serverName, err = serverFromInput(ctx, cf)
+		if err != nil {
+			return err
+		}
+	}
+	_, ok := cf.Servers[serverName]
+	if !ok {
+		return errors.New("Server not found")
+	}
+
+	if len(cf.Servers[serverName].Project) == 0 {
+		logrus.Warn("No context set; some commands will not work. Run 'rancher context switch'")
+	}
+
+	cf.CurrentServer = serverName
+	cf.Write()
+
+	return nil
+}
+
+// serverFromInput displays the list of servers from the local config and
+// prompt the user to select one.
+func serverFromInput(ctx *cli.Context, cf config.Config) (string, error) {
+
+	displayListServers(ctx, cf)
+	fmt.Print("Select a Server:")
+
+	reader := bufio.NewReader(os.Stdin)
+	input, err := reader.ReadString('\n')
```

Comment:
Lift [this](https://github.com/rancher/cli/blob/master/cmd/login.go#L228-L253) for use here. This will handle the case that the number is outside the range and inputs that aren't a number.

Reply:
Done